server/tpm2.go

```diff
@@ -84,10 +84,15 @@ func TPM2_Seal(data []byte) ([]byte, []byte, error) {
 	if sessHandle, _, err = tpm2.StartAuthSession(rwc, tpm2.HandleNull, tpm2.HandleNull, make([]byte, 16), nil, tpm2.SessionPolicy, tpm2.AlgNull, tpm2.AlgSHA256); err != nil {
 		return nil, nil, err
 	}
+	if *withpin {
```

Comment:
Shouldn't we check instead that pin != "" ? Why do we care about the flag at this level of abstraction?

Reply:
pin can be an empty string in two different scenarios:
- The `withpin` flag isn't set
- The `withpin` flag is set but the user provided an empty string when asked

Depending on the presence of this option, and independently of the provided PIN, we must set a password policy or not:

If the `withpin` option is provided when unsealing, the user (or an attacker) will input a PIN. If it doesn't match, two things can happend based on how the enrollment was done:
- if the `withpin` option was provided: The policies now matches, but the PIN doesn't -> the DA counter is incremented and can lock the TPM
- if the `withpin` option wasn't provided: The policies doesn't match because it was a password policy at enroll time and it's not one at unesal time. The provided PIN isn't even tested, and the DA counter isn't incremented. The user can now restart the attestation without the `withpin` option

I don't know if this is clear, we can call to talk about it further. Anyway, I understand this isn't documented well enough, but I really don't know how to make these things clear in code comments, maybe it'll be easier to extract it in some README.md and pinpoint it in the code?

Reply:
This makes sense to me. Please move the last paragraph of the function comment here, where it is easier to see what it refers to.